automata/fa/dfa.py

```diff
@@ -512,6 +531,88 @@ def isfinite(self):
         """
         Returns True if the DFA accepts a finite language, False otherwise.
         """
+        try:
+            return self.maximum_word_length() != float('inf')
+        except ValueError:
+            return True
+
+    def count_words_of_length(self, k):
+        """
+        Counts words of length k assuming state is the initial state
+        """
+        self._ensure_count_for_length(k)
+        return self._count_cache[k][self.initial_state]
+
+    def _ensure_count_for_length(self, k):
+        """
+        Populate count cache up to length k
+        """
+        while len(self._count_cache) <= k:
+            i = len(self._count_cache)
+            self._count_cache.append(defaultdict(int))
+            level = self._count_cache[i]
+            if i == 0:
+                for state in self.final_states:
+                    level[state] = 1
+            else:
+                prev_level = self._count_cache[i-1]
+                for state in self.states:
+                    state_count = 0
+                    for _, suffix_state in self.transitions[state].items():
+                        state_count += prev_level[suffix_state]
+                    level[state] = state_count
+ 
+    def words_of_length(self, k):
+        """
+        Generates all words of size k in the language represented by the DFA
+        """
+        self._ensure_words_of_length(k)
+        for word in self._word_cache[k][self.initial_state]:
+            yield word
+
+    def _ensure_words_of_length(self, k):
+        """
+        Populate word cache up to length k
+        """
+        while len(self._word_cache) <= k:
+            i = len(self._word_cache)
+            self._word_cache.append(defaultdict(set))
+            level = self._word_cache[i]
+            if i == 0:
+                for state in self.final_states:
```

Comment:
Maybe a bit extra, but I think you can do the inside of both of the loops with comprehensions.

Reply:
Like this? Or you had something else in mind?